src/SharedPipelineComponents/DataLockEvents/DeployDataLock/sql/dml/02 datalock.populate.reference.history.sql

```diff
@@ -1,53 +1,263 @@
-DECLARE @ProvidersToProcess TABLE (UKPRN bigint)
-INSERT INTO @ProvidersToProcess
-(UKPRN)
-SELECT
-	p.UKPRN
-FROM DataLock.vw_Providers p
-
-DECLARE @LastestDataLockEvents TABLE (EventId bigint)
+/* +++++ Monitoring setup */
+DECLARE @Now DATETIME 
+DECLARE @timestamp VARCHAR(255)
+DECLARE @duration INT
+DECLARE @operation NVARCHAR(255) = 'undefined'
+DECLARE @start NVARCHAR(255) = N'Start: %s %s.'
+DECLARE @finish NVARCHAR(255) = N'Finish: %s, took %d ms.'
+DECLARE @logsource varchar(max) = '02 datalock populate reference history'
+/* ----- Monitoring setup */
+
+/* +++++ start pre-op monitoring */
+SELECT @Now = GETDATE()
+SET @operation = 'Drop Target Table Keys, Constraints and Indexes'
+SET @timestamp = CONVERT(VARCHAR(255), @now, 121)
+RAISERROR (		@start,		10,		0,		@operation,		@timestamp		) WITH NOWAIT
+INSERT INTO [TestStack].[Logs] (	[LogId],	[Logger],	[LogLevel],	[EntryDate],	[Message],	[ErrorDetails]	)
+SELECT NEWID(),	@logsource,	0,	GETDATE(),	@operation + '- Started: [' + @timestamp + '] Duration: [' + cast(@duration AS VARCHAR(255)) + ']',	NULL
+/* ----- finish pre-op monitoring*/
+
+IF EXISTS (
+		SELECT [name]
+		FROM [sys].[indexes]
+		WHERE [name] = 'DataLockByStatusIdLearnRefNumberPriceEpisodeIdentifier'
+		)
+BEGIN
+	DROP INDEX [DataLockByStatusIdLearnRefNumberPriceEpisodeIdentifier]
+		ON [Reference].[DataLockEvents]
+END
+
+IF EXISTS (
+		SELECT [name]
+		FROM [sys].[indexes]
+		WHERE [name] = 'IX_Reference_DataLockEvents_UKPRN'
+		)
+BEGIN
+	DROP INDEX [IX_Reference_DataLockEvents_UKPRN]
+		ON [Reference].[DataLockEvents]
+END
+
+IF EXISTS (
+		SELECT [name]
+		FROM [sys].[key_constraints]
+		WHERE [name] = 'PK_Reference_DataLockEvents'
+		)
+BEGIN
+	ALTER TABLE [Reference].[DataLockEvents]
+	DROP CONSTRAINT [PK_Reference_DataLockEvents]
+END
+
+/* +++++ start post-op monitoring */
+SET @duration = DATEDIFF(MS, @Now, GETDATE())
+RAISERROR (		@finish,		10,		0,		@operation,		@duration		) WITH NOWAIT
+INSERT INTO [TestStack].[Logs] (	[LogId],	[Logger],	[LogLevel],	[EntryDate],	[Message],	[ErrorDetails]	)
+SELECT NEWID(), 	@logsource,	0,	GETDATE(),	@operation + '- Started: [' + @timestamp + '] Duration: [' + cast(@duration AS VARCHAR(255)) + ']',	NULL
+/* ----- finish post-op monitoring*/
+
+
+/* +++++ start pre-op monitoring */
+SELECT @Now = GETDATE()
+SET @operation = 'Establish Provider scope'
+SET @timestamp = CONVERT(VARCHAR(255), @now, 121)
+RAISERROR (		@start,		10,		0,		@operation,		@timestamp		) WITH NOWAIT
+INSERT INTO [TestStack].[Logs] (	[LogId],	[Logger],	[LogLevel],	[EntryDate],	[Message],	[ErrorDetails]	)
+SELECT NEWID(),	@logsource,	0,	GETDATE(),	@operation + '- Started: [' + @timestamp + '] Duration: [' + cast(@duration AS VARCHAR(255)) + ']',	NULL
+/* ----- finish pre-op monitoring*/
+
+DECLARE @ProvidersToProcess TABLE (
+	[UKPRN] BIGINT
+	)
+
+INSERT INTO @ProvidersToProcess (
+	[UKPRN]
+	)
+SELECT [UKPRN]
+FROM [DataLock].[vw_Providers]
```

Comment:
Does this temp table add any value?  Couldn't we just join directly to the view?

Reply:
Yes, a further refactor could remove this.